// ng-pipe-filter/esm5/lib/pipes/object-index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { ToArrayPipe } from './object/to-array.pipe';
import { ToKeyValPipe } from './object/to-key-val.pipe';
import { ToKeyPipe } from './object/to-key.pipe';
export { ToArrayPipe } from './object/to-array.pipe';
export { ToKeyValPipe } from './object/to-key-val.pipe';
export { ToKeyPipe } from './object/to-key.pipe';
var ObjectPipesModule = /** @class */ (function () {
    function ObjectPipesModule() {
    }
    ObjectPipesModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        ToArrayPipe,
                        ToKeyValPipe,
                        ToKeyPipe
                    ],
                    exports: [
                        ToArrayPipe,
                        ToKeyValPipe,
                        ToKeyPipe
                    ]
                },] }
    ];
    return ObjectPipesModule;
}());
export { ObjectPipesModule };
