// ng-pipe-filter/esm2015/lib/pipes/string-index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommaAddPipe } from './string/comma-add.pipe';
import { CommaRemovePipe } from './string/comma-remove.pipe';
import { Nl2brPipe } from './string/nl2br.pipe';
import { StripTagsPipe } from './string/strip-tags.pipe';
export { CommaAddPipe } from './string/comma-add.pipe';
export { CommaRemovePipe } from './string/comma-remove.pipe';
export { Nl2brPipe } from './string/nl2br.pipe';
export { StripTagsPipe } from './string/strip-tags.pipe';
export class StringPipesModule {
}
StringPipesModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    CommaAddPipe,
                    CommaRemovePipe,
                    StripTagsPipe,
                    Nl2brPipe,
                ],
                exports: [
                    CommaAddPipe,
                    CommaRemovePipe,
                    StripTagsPipe,
                    Nl2brPipe,
                ]
            },] }
];
